gobblin-data-management/src/main/java/org/apache/gobblin/data/management/trash/TimeBasedSnapshotCleanupPolicy.java

```diff
@@ -41,6 +42,7 @@ public TimeBasedSnapshotCleanupPolicy(Properties props) {
   @Override
   public boolean shouldDeleteSnapshot(FileStatus snapshot, Trash trash) {
     DateTime snapshotTime = Trash.TRASH_SNAPSHOT_NAME_FORMATTER.parseDateTime(snapshot.getPath().getName());
-    return snapshotTime.plusMinutes(this.retentionMinutes).isBeforeNow();
+    // To ensure that the comparison between snapshotTime and the current time is done in the same time zone
+    return snapshotTime.plusMinutes(this.retentionMinutes).isBefore(DateTime.now(DateTimeZone.UTC));
```

Comment:
Should there be a configuration to ensure that it is using UTC compared to the old behavior? You don't want existing retention jobs to have unintended consequences due to change in behavior that is not explicit

Reply:
I think it would be more extensible if we had something like the following:
```
public static final String RETENTION_SNAPSHOT_TIMEZONE = "gobblin.trash.snapshot.retention.timezone"
Timezone retentionSnapshotTimezone;
...
this.retentionSnapshotTImezone = props.containsKey(RETENTION_SNAPSHOT_TIMEZONE) ? Timezone.getTimeZone(props.getProperty(RETENTION_SNAPSHOT_TIMEZONE)) : DateTimeZone.getDefault();
...
return snapshotTime.plusMinutes(this.retentionMinutes).isBefore(DateTime.now(this.retentionSnapshotTimezone));
```
This allows the retention snapshot be more configurable in the long term and more explciit what the behavior is.

Reply:
thanks for the suggestion, this is indeed much cleaner and more reasonable